### ISSUE TITLE
Getting cuda OOM at imgsz 1280 and batch size 1

### ISSUE THREAD

**Issue by chandan-wiai** [ultralytics/ultralytics#20853]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi all,

I am trying to train a segmentation model using yolov9e-seg model. It's a small object segmentation task with around 400-800 objects per image. I am using L4 GPUs (24 GB GPU RAM) for training.
As this is a small object segmentation task, I have seen a consistent improvement in performance when I increase the imgsz parameter from 640 to 960. I want to train the model at even higher resolutions such as 1120, 1280, 1600 etc. 

On a single L4 GPU, I am able to fit a batch size of 2 at 960 imgsz. But at imgsz 1120 and 1280 I get cuda OOM error even at batch size of 1. I don't understand why this is happening. Usually, required gpu memory is proportional to the square of imgsz parameter. By this logic the memory required for 1120 resolution should only increase by 36% ( (1120 * 1120) / (960 * 960))  and reducing batch size by half (from 2 to 1) should definitely be able to accommodate this additional required space in GPU RAM. 
Any suggestions so that I can fit a batch size of 1 at imgsz 1280 somehow? Thanks.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @chandan-wiai, thank you for reaching out and providing detailed context about your segmentation training scenario 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for further guidance, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. You may also find answers to similar questions there.

As this may be a 🐛 Bug Report related to CUDA OOM, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if possible, including your exact training command, environment details, and any relevant logs. This will help us better diagnose the GPU memory issue you’re experiencing.

If you're customizing your training, sharing additional information such as sample dataset images and full training logs can also assist us in understanding your setup. Don't forget to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for further optimization strategies.

Join the Ultralytics community for real-time support or to discuss with others:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies and [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help you get started—an Ultralytics engineer will assist you here soon! 🛠️
